mesonbuild/utils/universal.py

```diff
@@ -472,8 +473,8 @@ def get_compiler_for_source(compilers: T.Iterable['Compiler'], src: 'FileOrStrin
     raise MesonException(f'No specified compiler can handle file {src!s}')
 
 
-def classify_unity_sources(compilers: T.Iterable['Compiler'], sources: T.Sequence['FileOrString']) -> T.Dict['Compiler', T.List['FileOrString']]:
-    compsrclist: T.Dict['Compiler', T.List['FileOrString']] = {}
+def classify_unity_sources(compilers: T.Iterable[Compiler], sources: T.Iterable[FileTypes]) -> T.Dict[Compiler, T.List[FileTypes]]:
```

Comment:
This patch is itself the thing I stated must be destroyed.

Reply:
`--verbose`, you mean that `str` is `T.Iterable[str]` just like it is `T.Sequence[str]`? This approach is still preferrable to yours if changed to have `T.List[FileTypes]` as the argument:

```suggestion
def classify_unity_sources(compilers: T.Iterable[Compiler], sources: T.List[FileTypes]) -> T.Dict[Compiler, T.List[FileTypes]]:
```

I just checked and it still passes run_mypy. I'm opening a separate PR.

Reply:
I know but I disagree that Union is the right solution.

Ok, won't open a PR but you can pick it yourself at github.com/bonzini/meson/commit/8e072288ffa0bca7e3c338d6ec5ca4588e5e8e0d.patch